### PR TITLE
Add Zig 0.12.0

### PR DESCRIPTION
**Pull request by verhovsky**

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -2926,6 +2926,8 @@ compiler.zcxx0100.exe=/opt/compiler-explorer/zig-0.10.0/zig
 compiler.zcxx0100.semver=0.10.0
 compiler.zcxx0110.exe=/opt/compiler-explorer/zig-0.11.0/zig
 compiler.zcxx0110.semver=0.11.0
+compiler.zcxx0110.exe=/opt/compiler-explorer/zig-0.12.0/zig
```
Comment (partouf) [inline etc/config/c++.amazon.properties:2929]: should be `zcxx0120`

File: etc/config/c.amazon.properties [hunk 2]
```diff
@@ -2712,6 +2712,8 @@ compiler.zcc0100.exe=/opt/compiler-explorer/zig-0.10.0/zig
 compiler.zcc0100.semver=0.10.0
 compiler.zcc0110.exe=/opt/compiler-explorer/zig-0.11.0/zig
 compiler.zcc0110.semver=0.11.0
+compiler.zcc0110.exe=/opt/compiler-explorer/zig-0.12.0/zig
```
Comment (partouf) [inline etc/config/c.amazon.properties:2715]: should be `zcc0120`

**Comment by verhovsky** [comment]
When will it appear on https://godbolt.org/ ?

**Comment by partouf** [comment]
this is now live

**Pull request merged**